troi/patches/ab_similar_recordings.py

```diff
@@ -0,0 +1,42 @@
+import troi.filters
+import troi.listenbrainz.recs
+import troi.musicbrainz.recording_lookup
+from troi.acousticbrainz.annoy import AnnoyLookupElement
+
+
+VALID_METRICS = ['mfccs', 'mfccsw', 'gfccs', 'gfccsw', 'key', 'bpm', 'onsetrate', 'moods',
+                 'instruments', 'dortmund', 'rosamerica', 'tzanetakis']
+
+
+class ABSimilarRecordingsPatch(troi.patch.Patch):
+
+    @staticmethod
+    def inputs():
+        return [
+            {"type": str, "name": "recording_id", "desc": "MusicBrainz recording id", "optional": False},
+            {"type": str, "name": "similarity_type", "desc": "Type of acousticbrainz similarity", "optional": False}]
+
+    @staticmethod
+    def slug():
+        return "ab-similar-recordings"
+
+    @staticmethod
+    def description():
+        return "Find acoustically similar recordings from AcousticBrainz"
+
+    def create(self, inputs):
+        recording_id = inputs[0]
+        similarity_type = inputs[1]
+
+        if similarity_type not in VALID_METRICS:
```

Comment:
Maybe make this check case insensitive?

Reply:
sure. do you think this check makes more sense here or in `AnnoyLookupElement`? (or maybe in both?)

Reply:
Both seems overkill, but in the Element makes sense to me.

Reply:
added this, and moved check to the element